app/main/views/frameworks.py

```diff
@@ -835,133 +828,32 @@ def framework_agreement(framework_slug):
             date_parse(supplier_framework['agreementReturnedAt'])
         )
 
-    # if there's a frameworkAgreementVersion key, it means we're on G-Cloud 8 or higher
-    if framework.get('frameworkAgreementVersion'):
-        def lot_result(drafts_for_lot):
-            if any(draft['status'] == 'submitted' for draft in drafts_for_lot):
-                return 'Successful'
-            elif any(draft['status'] == 'failed' for draft in drafts_for_lot):
-                return 'Unsuccessful'
-            else:
-                return 'No application'
-
-        drafts, complete_drafts = get_drafts(data_api_client, framework_slug)
-        complete_drafts_by_lot = {
-            lot['slug']: [draft for draft in complete_drafts if draft['lotSlug'] == lot['slug']]
-            for lot in framework['lots']
-        }
-        lot_results = {k: lot_result(v) for k, v in complete_drafts_by_lot.items()}
+    def lot_result(drafts_for_lot):
+        if any(draft['status'] == 'submitted' for draft in drafts_for_lot):
+            return 'Successful'
+        elif any(draft['status'] == 'failed' for draft in drafts_for_lot):
+            return 'Unsuccessful'
+        else:
+            return 'No application'
 
-        return render_template(
-            'frameworks/contract_start.html',
-            signature_page_filename=SIGNATURE_PAGE_FILENAME,
-            framework=framework,
-            framework_urls=content_loader.get_message(framework_slug, 'urls'),
-            lots=[{
-                'name': lot['name'],
-                'result': lot_results[lot['slug']]
-            } for lot in framework['lots']],
-            supplier_framework=supplier_framework,
-            supplier_registered_name=get_supplier_registered_name_from_declaration(supplier_framework['declaration']),
-        ), 200
+    drafts, complete_drafts = get_drafts(data_api_client, framework_slug)
+    complete_drafts_by_lot = {
+        lot['slug']: [draft for draft in complete_drafts if draft['lotSlug'] == lot['slug']]
+        for lot in framework['lots']
+    }
+    lot_results = {k: lot_result(v) for k, v in complete_drafts_by_lot.items()}
```

Comment:
Readability could benefit from (IMO)
```
lot_results = {lot_slug: lot_result(draft_services) for lot_slug, draft_services in complete_drafts_by_lot.items()}
```
I think this has the advantage of cementing the structure of the above comprehension. But it's a bit of a nitpick tbh :)